packages/eslint-plugin/tests/fixtures/tsconfig.noUncheckedIndexedAccess.json

```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "noUncheckedIndexedAccess": true
```

Comment:
Really nice find here. I shudder to think what other subtle edge cases have popped up because of that 😄

Reply:
fwiw this issue can arise even without `noUncheckedIndexAccess`, but off the top of my head I can't think of an example that isn't totally contrived :-).

Reply:
Oh? I was playing around with this for a good bit to try to find one but couldn't find anything. Is there a code snippet you have in mind?

Reply:
I couldn't find one either to be honest, but I didn't try all that hard. Maybe it really is dependent on the flag. But even if it isn't I think this fix should suffice. I'll do some experimentation.